### ISSUE TITLE
 Instead of picking up the metric, hyperparam search is using SamplePerSeconds to select the best model

### ISSUE THREAD

**Issue by budhiraja** [huggingface/notebooks#13]
Hi, 
I am trying to replicate the notebook on this path : [https://github.com/huggingface/notebooks/blob/master/examples/text_classification.ipynb](https://github.com/huggingface/notebooks/blob/master/examples/text_classification.ipynb)

But in my run, instead of selecting the model using mathews correlation, its selecting a model based on the epoch running time. For example, look at the log printed in the notebook which I run: `Trial 0 finished with value: 1645.5148768624724 and parameters: {'learning_rate': 2.0970346847322057e-05, 'num_train_epochs': 5, 'seed': 35, 'per_device_train_batch_size': 32}. Best is trial 0 with value: 1645.5148768624724.`

Interestingly, the tables printed in my notebook have two additional columns: RunTime and SamplesPerSeconds. My guess is the library code is picking up the last column of the dataframe and not the one with the name of the metric. Adding a snapshot of the table below.

| Epoch | Training Loss | Validation Loss | Matthews Correlation |  Runtime | Samples Per Second |
|------:|--------------:|----------------:|---------------------:|---------:|-------------------:|
|     1 |        No log |        0.482555 |             0.435778 | 0.631800 |        1650.953000 |
|     2 |      0.450100 |        0.494479 |             0.488171 | 0.632700 |        1648.565000 |
|     3 |      0.450100 |        0.574674 |             0.510249 | 0.637900 |        1635.140000 |
|     4 |      0.218400 |        0.637276 |             0.519209 | 0.643200 |        1621.627000 |
|     5 |      0.218400 |        0.680201 |             0.520577 | 0.634300 |        1644.360000 |

Can someone please help me with whether I am missing something?

**Comment by budhiraja** [comment]
@sgugger @julien-c can you please help?

**Comment by sgugger** [comment]
What's your version of transformers? I think this bug has been fixed.

**Comment by budhiraja** [comment]
I am using `4.3.3` and I also tried `pip install transformers --upgrade` but it is the same version.

**Comment by sgugger** [comment]
Can you try with an install from source?
```
pip install git+https://github.com/huggingface/transformers
```

**Comment by cb-podsights** [comment]
@sgugger I am experiencing this issue as well and I am running 4.5.0. Here's my code:

train_args = TrainingArguments(
                                output_dir="huggingface_output_1",
                                metric_for_best_model="recall",
                                load_best_model_at_end=True,
                                evaluation_strategy='steps',
                                eval_steps=500,
                                do_eval=True,
                                )

trainer = Trainer(
        model_init=hp_model_init,
        args=train_args,
        train_dataset=ds["train"],
        eval_dataset=ds["eval"],
        compute_metrics=compute_metrics,
    )
    
    model = trainer.hyperparameter_search(
        direction="maximize",
        n_trials=10,
        hp_space=my_hp_space,
        backend="optuna",
        compute_objective=compute_recall
    )

Am I setting something incorrectly or failing to set something, or is this a bug?
